zorg/buildbot/builders/annotated/libc-linux.py

```diff
@@ -67,17 +63,12 @@ def main(argv):
         # explicit here, which reduces one step of setting environment
         # variables when setting up workers.
         cmake_args = ['-GNinja',
+                      '-DLLVM_ENABLE_RUNTIMES=libc',
                       '-DCMAKE_C_COMPILER=%s' % cc,
                       '-DCMAKE_CXX_COMPILER=%s' % cxx]
         if lint_build:
             cmake_args.append('-DLLVM_LIBC_CLANG_TIDY=%s' % clang_tidy)
 
-        if runtimes_build:
-          projects = ['llvm', 'clang']
-          cmake_args.append('-DLLVM_ENABLE_RUNTIMES=libc')
-        else:
-          projects = ['llvm', 'libc']
-
```

Comment:
I think you still need `llvm` in projects because the cmake command is rooted in the `llvm` directory. See `run_command(['cmake', os.path.join(source_dir, 'llvm')] + cmake_args)`

Reply:
Have you verified that? I've been doing runtimes builds (rooted in the runtimes dir) locally without ANY value set for `-DLLVM_ENABLE_PROJECTS` for building+testing libc for a while now.  Even when I did core clang+llvm work, I think would use the llvm/ dir as the root and only explicitly list clang and lld in my `-DLLVM_ENABLE_PROJECTS`. So I would be surprised if llvm needed to appear explicitly in `-DLLVM_ENABLE_PROJECTS` when using llvm/ dir as the root for cmake (for llvm-libc, or any other sub project).

---

From a recent team discussion though, @lntue and @michaelrj-google mentioned that the use of llvm/ dir as the root and clang in the projects list will do a _bootstrap_ build.

@lntue also pointed out this this is how we're testing fixed point support continuously, since the old clang most other buildbots are running are too old for fixed point support. Landing this PR right now would regress our CI coverage for fixed point.

So I think it's important to distinguish "bootstrap" builds (i.e. build clang first, then llvm-libc with that clang) from "runtimes" builds (anchored in runtimes/ for cmake).  Those are two distinct things to me which are currently being conflated by the term "runtimes."

I have near term goals to move all build bots to newer versions of clang.  That alone will resolve having coverage for fixed point.

It may still be handy to have "bootstrap" builds in CI, though they will take much longer to run since they need to build all of llvm and clang first.

Next steps to me then are perhaps:
1. rename debian-dbg-runtimes-build to debian-dbg-bootstrap-build (rather than spin down debian-dbg-runtimes-build as I had originally intended in https://github.com/llvm/llvm-project/issues/78479#issuecomment-2492580339.  This splits the conflation of runtimes with runtimes/ cmake root from "bootstrap" builds.  Probably need to update buildbot config files, move/rename dirs, and wipe build dirs.
2. update this PR to move remaining builds to LLVM_ENABLE_RUNTIMES

Reply:
SGTM

Reply:
https://github.com/llvm/llvm-zorg/pull/332 as precursor so that 1 doesn't turn off any CI builds.

Reply:
https://lab.llvm.org/buildbot/#/builders?tags=libc now shows that `libc-x86_64-debian-dbg-runtimes-build` has been renamed to `libc-x86_64-debian-dbg-bootstrap-build`. I've rebased this and updated it, so this thread should be resolved. Ready for rereview.